src/shared/util.js

```diff
@@ -907,58 +903,37 @@ var PageViewport = (function PageViewportClosure() {
 
     // creating transform to convert pdf coordinate system to the normal
     // canvas like coordinates taking in account scale and rotation
-    var centerX = (viewBox[2] + viewBox[0]) / 2;
-    var centerY = (viewBox[3] + viewBox[1]) / 2;
-    var rotateA, rotateB, rotateC, rotateD;
     rotation = rotation % 360;
-    rotation = rotation < 0 ? rotation + 360 : rotation;
-    switch (rotation) {
-      case 180:
-        rotateA = -1; rotateB = 0; rotateC = 0; rotateD = 1;
-        break;
-      case 90:
-        rotateA = 0; rotateB = 1; rotateC = 1; rotateD = 0;
-        break;
-      case 270:
-        rotateA = 0; rotateB = -1; rotateC = -1; rotateD = 0;
-        break;
-      // case 0:
-      default:
-        rotateA = 1; rotateB = 0; rotateC = 0; rotateD = -1;
-        break;
+    if (rotation < 0) {
+      rotation = rotation + 360;
     }
-
-    if (dontFlip) {
-      rotateC = -rotateC; rotateD = -rotateD;
-    }
-
-    var offsetCanvasX, offsetCanvasY;
+    var sideways = rotation % 180;
+    var a = rotation < 180 ? 1 : -1;
```

Comment:
My Proposal: 
```
    var scaleX = rotation < 180 ? scale : -scale;
    var scaleY = dontFlip ? scaleX : -scaleX;
    var rotate = sideways ? [0, scaleX, -scaleY, 0] : [scaleX, 0, 0, scaleY];

...

    this.transform = rotate.concat(
      offsetX + (width / 2 - rotate[0] * centerX - rotate[2] * centerY),
      offsetY + (height / 2 - rotate[1] * centerX - rotate[3] * centerY)
    );
```
which you get by the following refactoring steps:
1. multiply `a` by `scale`, and remove all ` * scale` in the `this.transform = ...` statement
2. multiply `flip` by `a`, remove `a *` in the `var rotate = ...` statement
3. rename `a` to `scaleX`
4. rename `flip` to `scaleY`
5. flip sign of `scaleY`
6. move `offsetX/Y +` to the beginning of the respective lines to have more visual alignement